go/tasks/plugins/webapi/snowflake/plugin.go

```diff
@@ -0,0 +1,294 @@
+	return "default", p.cfg.ResourceConstraints, nil
+}
+
+func (p Plugin) Create(ctx context.Context, taskCtx webapi.TaskExecutionContextReader) (webapi.ResourceMeta,
+	webapi.Resource, error) {
+	task, err := taskCtx.TaskReader().Read(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+	fmt.Print("kevin = ", p.cfg.TokenKey)
+	token, err := taskCtx.SecretManager().Get(ctx, p.cfg.TokenKey)
+	if err != nil {
+		return nil, nil, err
+	}
+	custom := task.GetCustom()
+	snowflakeQuery := QueryJobConfig{}
+	err = pluginUtils.UnmarshalStructToObj(custom, &snowflakeQuery)
+	if err != nil {
+		return nil, nil, errors.Wrapf(ErrUser, err, "Expects a valid PrestoQuery proto in custom field.")
+	}
+	outputs, err := template.Render(ctx, []string{
+		snowflakeQuery.Account,
+		snowflakeQuery.Warehouse,
+		snowflakeQuery.Schema,
+		snowflakeQuery.Database,
+		snowflakeQuery.Statement,
+	}, template.Parameters{
+		TaskExecMetadata: taskCtx.TaskExecutionMetadata(),
+		Inputs:           taskCtx.InputReader(),
+		OutputPath:       taskCtx.OutputWriter(),
+		Task:             taskCtx.TaskReader(),
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+	queryInfo := QueryInfo{
+		Account:   outputs[0],
+		Warehouse: outputs[1],
+		Schema:    outputs[2],
+		Database:  outputs[3],
+		Statement: outputs[4],
+	}
+
+	if len(queryInfo.Warehouse) == 0 {
+		queryInfo.Warehouse = p.cfg.DefaultWarehouse
+	}
+	req, err := buildRequest(post, queryInfo, p.cfg.snowflakeEndpoint,
+		queryInfo.Account, token, "", false)
+	if err != nil {
+		return nil, nil, err
+	}
+	resp, err := p.client.Do(req)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer resp.Body.Close()
+	data, err := buildResponse(resp)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	queryID := fmt.Sprintf("%v", data["statementHandle"])
+	message := fmt.Sprintf("%v", data["message"])
+
+	return &ResourceMetaWrapper{queryID, queryInfo.Account, token},
+		&ResourceWrapper{StatusCode: resp.StatusCode, Message: message}, nil
+}
+
+func (p Plugin) Get(ctx context.Context, taskCtx webapi.GetContext) (latest webapi.Resource, err error) {
+	exec := taskCtx.ResourceMeta().(*ResourceMetaWrapper)
+	req, err := buildRequest(get, QueryInfo{}, p.cfg.snowflakeEndpoint,
+		exec.Account, exec.Token, exec.QueryID, false)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := p.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	data, err := buildResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+	message := fmt.Sprintf("%v", data["message"])
+	return &ResourceWrapper{
+		StatusCode: resp.StatusCode,
+		Message:    message,
+	}, nil
+}
+
+func (p Plugin) Delete(ctx context.Context, taskCtx webapi.DeleteContext) error {
+	exec := taskCtx.ResourceMeta().(*ResourceMetaWrapper)
+	req, err := buildRequest(post, QueryInfo{}, p.cfg.snowflakeEndpoint,
+		exec.Account, exec.Token, exec.QueryID, true)
+	if err != nil {
+		return err
+	}
+	resp, err := p.client.Do(req)
```

Comment:
If this returns a non-200, is it an err @EngHabu ?  Or is it a response with no error but a non-200?  do we need to check for it?